src/managers/validatedDynamodbManager.ts

```diff
@@ -241,6 +241,61 @@ function validateKeyConditionExpression<T extends object>(type:{new(...args: any
     }
 }
 
+function validateDeleteConditionExpression<T extends object>(type:{new(...args: any[]):T}, param:{conditionExpression?:string, expressionAttributeValues?:object, expressionAttributeNames?:object})
+{
+    if(param === undefined) return;
+    
+    let instance = new type();
+    let hashes = Reflector.getAllHashKeys(instance).map(hash => Key.parse(hash).propertyName);
+    let ranges = Reflector.getAllRangeKeys(instance).map(range => Key.parse(range).propertyName);
+    let columns = Reflector.getColumns(instance).map(column => Key.parse(column).propertyName);
+    
+    //Add hash/rane/id column here because expressionAttributeNames may include one of those from keyConditions expression.
+    columns = [...columns, Const.HashColumn, Const.RangeColumn, Const.IdColumn];
```

Comment:
it looks like no,  I don't think we need the hash range or id either because DynamoDbManager   does not include key conditions.

```
 async delete<T extends object>(type:{new(...args: any[]):T}, id:string|number,  params?:{conditionExpression:string, expressionAttributeValues?:object, expressionAttributeNames?:object}, transaction?:DynamoDbTransaction, autoCommit:boolean = true): Promise<void>
```